ci/Jenkinsfile

```diff
@@ -0,0 +1,100 @@
+@Library('jenkins_shared_lib') _
+
+pipeline {
+  agent any
+  environment {
+    BUILD_TYPE = sh(returnStdout: true, script: 'rapids-build-type')
+  }
+  options {
+    buildDiscarder(logRotator(numToKeepStr: '10'))
+    ansiColor('xterm')
+    disableConcurrentBuilds(abortPrevious: true)
+  }
+  stages {
+    stage("Style Checks") {
+      steps {
+        sh "env"
+      }
+    }
+    stage("C++ Matrix Build") {
+      environment {
+        PARALLEL_LEVEL = '4'
+      }
+      options {
+        timeout(time: 30, unit: 'MINUTES', activity: true)
+      }
+      steps {
+        script {
+          // generate the CUDA stages across different configuration options
+          parallel generateStage("cuda_build", {
+            sh "./ci/build_cpp.sh"
+          })
+        }
+      }
+    }
+
+    stage("Python Matrix Build") {
+      options {
+        timeout(time: 30, unit: 'MINUTES', activity: true)
+      }
+      steps {
+        script {
+          // generate the python stages across different configuration options
+          parallel generateStage("python_build", {
+            sh "./ci/build_py.sh"
+          })
+        }
+      }
+    }
+
+    stage("PR/Branch Test Stage") {
+      when { not { triggeredBy 'TimerTrigger' } }
+      options {
+        timeout(time: 10, unit: 'MINUTES', activity: true)
+      }
+      steps {
+        script {
+          // generate the PR/Branch test stages across different configuration options
+          parallel generateStage("branch_pr_test",{
+            sh "./ci/gpu_test.sh"
+          })
+        }
+      }
+    }
+
+    stage("Nightly Test Stage") {
+      when { triggeredBy 'TimerTrigger' }
+      options {
+        timeout(time: 10, unit: 'MINUTES', activity: true)
+      }
+      environment { UPLOAD_PKGS = "true" }
+      steps {
+        script {
+          // generate the nightly test stages across different configuration options
+          parallel generateStage("branch_pr_test",{
+            sh "./ci/gpu_test.sh"
+          })
+        }
+      }
+    }
+
+    stage("Upload Anaconda Packages") {
+      when {
+        anyOf {
+          environment name: 'BUILD_TYPE', value: 'branch'
+          environment name: 'BUILD_TYPE', value: 'nightly'
+        }
+      }
+
+      steps {
+        sh '''#!/bin/bash
+          CPP_CHANNEL=$(rapids-download-conda-from-s3 cpp)
+          PYTHON_CHANNEL=$(rapids-download-conda-from-s3 python)
+
+          rapids-upload-to-anaconda "${CPP_CHANNEL}"
+          rapids-upload-to-anaconda "${PYTHON_CHANNEL}"
+        '''
```

Comment:
Let's move these lines to their own script for consistency with the rest of the pipeline.